other/scoring_algorithm.py

```diff
@@ -72,14 +72,45 @@ def procentual_proximity(
 
         score_lists.append(score)
 
+    return score_lists
+
+
+def generate_final_scores(score_lists: list[list[float]]) -> list[float]:
+    """
+    >>> generate_final_scores([[1.0, 0.0, 0.33333333333333337],
+    ...                        [0.75, 0.0, 1.0],
+    ...                        [0.25, 1.0, 0.0]])
+    [2.0, 1.0, 1.3333333333333335]
+    """
     # initialize final scores
     final_scores: list[float] = [0 for i in range(len(score_lists[0]))]
 
-    # generate final scores
     for slist in score_lists:
         for j, ele in enumerate(slist):
             final_scores[j] = final_scores[j] + ele
 
+    return final_scores
+
+
+def procentual_proximity(
+    source_data: list[list[float]], weights: list[int]
+) -> list[list[float]]:
+    """
+    weights - int list
+    possible values - 0 / 1
+    0 if lower values have higher weight in the data set
+    1 if higher values have higher weight in the data set
+
+    >>> procentual_proximity([[20, 60, 2012],[23, 90, 2015],[22, 50, 2011]], [0, 0, 1])
+    [[20, 60, 2012, 2.0], [23, 90, 2015, 1.0], [22, 50, 2011, 1.3333333333333335]]
+    """
+
+    data_lists: list[list[float]] = get_data(source_data)
+
+    score_lists: list[list[float]] = calculate_each_score(data_lists, weights)
+
+    final_scores: list[float] = generate_final_scores(score_lists)
+
```

Comment:
```suggestion
    data_lists = get_data(source_data)
    score_lists = calculate_each_score(data_lists, weights)
    final_scores = generate_final_scores(score_lists)

```
There's no need to specify variable type hint as the functions are already annotated.

Reply:
Fixed